### PR TITLE
Separate concerns in useNavigationBlocks

### PR DESCRIPTION
**Pull request by adamziel**
## Description
This is a follow-up on https://github.com/WordPress/gutenberg/pull/22821

This PR refactors `useNavigationBlocks` so that it does only what the label says - returns navigation blocks. No more fetching or saving logic embedded. Every part has it's own dedicated hook, and they're all wired together in the `useMenuEditor` effect.

## How has this been tested?
1. Enable the navigation experiment in Gutenberg > Experiments
1. Go to Gutenberg > Navigation (beta)
1. Add a few menu items, update a few other, delete a few existing ones, then hit save
1. Confirm everything was saved and worked properly
1. In general - confirm nothing is broken

## Types of changes
Non-breaking changes

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-navigation/src/components/menu-editor/use-create-missing-menu-items.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		for ( const { clientId, name } of flattenBlocks( blocks ) ) {
+			// No need to create menuItems for the wrapping navigation block
+			if ( name === 'core/navigation' ) {
+				continue;
+			}
+			// Menu item was already created
+			if ( clientId in menuItemsRef.current ) {
+				continue;
+			}
+			// Already in the queue
```
Comment (draganescu) [inline packages/edit-navigation/src/components/menu-editor/use-create-missing-menu-items.js:36]:
```suggestion
			// Menu item already in the queue
```

File: packages/edit-navigation/src/components/menu-editor/use-create-missing-menu-items.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		}
+	};
+	const onCreated = useCallback(
+		( callback ) => promiseQueueRef.current.then( callback ),
+		[ promiseQueueRef.current ]
+	);
+	return { createMissingMenuItems, onCreated };
+}
+
+async function createDraftMenuItem() {
```
Comment (draganescu) [inline packages/edit-navigation/src/components/menu-editor/use-create-missing-menu-items.js:58]: This is `async` but we don't `await` anything.

File: packages/edit-navigation/src/components/menu-editor/use-navigation-blocks.js [hunk 3]
```diff
@@ -168,9 +79,7 @@ const menuItemsToLinkBlocks = (
 	};
 
 	// menuItemsToTreeOfLinkBlocks takes an array of top-level menu items and recursively creates all their innerBlocks
-	const linkBlocks = menuItemsToTreeOfLinkBlocks(
-		itemsByParentID[ 0 ] || []
-	);
+	const linkBlocks = menuItemsToTreeOfBlocks( itemsByParentID[ 0 ] || [] );
```
Comment (draganescu) [inline packages/edit-navigation/src/components/menu-editor/use-navigation-blocks.js:82]: We might want to rename `linkBlocks` on the basis that in the future we could have other kinds of blocks coming in.

**Review by draganescu — CHANGES_REQUESTED** [review]
Everything is minor and I'd approve but there appears to be one rogue `async` that has to be changed.

**Comment by adamziel** [comment]
@draganescu I just addressed all your notes, thank you!

**Review by draganescu — APPROVED** [review]
I tested this and it breaks nothing, while the code looks pretty much better. Great refactoring @adamziel

**Pull request merged**